### DISCUSSION TITLE
onModel* vs onRecord*Request?

### DISCUSSION THREAD

**Discussion by etfz** [pocketbase/pocketbase#5427] [Q&A] [answered]
Hi,

I'm looking to monitor record changes and additions. Both `onModelAfterUpdate` and `onRecordAfterUpdateRequest` are triggered when modifying a record. When would I want to use one or the other?

**Comment by ganigeorgiev** [comment, marked as answer]
Please see https://github.com/pocketbase/pocketbase/discussions/5264, https://github.com/pocketbase/pocketbase/discussions/4966.

As mentioned in the documentation of the related methods and the linked posts, the `*Request` suffixed hooks fire only when sending requests to the related Web API endpoint. The `OnModel*` fires for any DB model operation executed through the `app.Dao()` (a single request can fire one or many `OnModel*` hooks depending on how many records are being created/modified).
